lib/ansible/module_utils/facts/hardware/linux.py

```diff
@@ -524,6 +524,16 @@ def get_mount_info(self, mount, device, uuids):
 
         return mount_size, uuid
 
+    @staticmethod
+    def _is_blocking_mount(device, fstype):
+        # The only mounts starting without slashes should be network mounts, special filesystems
+        # and ZFS (which we want to keep)
+        if not device.startswith(('/', '\\')) and fstype != 'zfs':
```

Comment:
While you are here, can you please add support for

-  https://github.com/ansible/ansible/issues/24644 
-  https://github.com/ansible/ansible/issues/41494

Reply:
The question is then, whether ansible_mounts should return network mounts or not. According to @samdoran, the original purpose of the conditional was to specifically not return network mounts. But the filesystems in the bug reports are mostly network ones (glusterfs, gpfs, nfs).

Also, where should we draw the line between regular filesystems and special ones. Right now, no special filesystems are listed. But should we list tmpfs (including /run etc.), as someone requested in #24644? (Bad idea in my opinion.)

Reply:
After some time off and looking at this with fresh eyes, plus reading through related issues, my thinking on this has evolved a bit.

The intention was to filter out most special filesystems (things that don't start with `/`) since that tends to be a lot of mounts. It then expanded to excluding network mounts because that could cause fact gathering to hang (https://github.com/ansible/ansible/pull/41502#issuecomment-443295416). A potential workaround for that was added in #49398, but I don't think it accomplishes the original intent. A timeout decorator is in the works that would solve the problem correctly (#72463).

Reply:
I think this should be a mount filtering function that has the goal of excluding mounts most special mounts (not starting with `/`) and allowing certain mount types.

So two things needs to happen here:

1) We need a filtering function whose job is to include the mounts we feel should show up in mount facts
2) If we want to include network mounts (which there is a lot of desire for), we need to extract the [code that processes each mount](https://github.com/ansible/ansible/blob/3e9943bc5e7a9cd393757aa8100d7fed80bd316e/lib/ansible/module_utils/facts/hardware/linux.py#L567-L593) into a standalone method that can use the existing `timeout()` decorator from `ansible.module_utils.facts.timeout`. (some solution mention in https://github.com/ansible/ansible/pull/41502#issuecomment-443295416).

Regarding the filter function, I don't think it should include special file systems like `tmpfs`, `sysfs`, `cgroup`. Assuming we use the timeout decorator, I would be ok adding `fusefs`, `nfs`, and `gpfs`

I found some [example `gpfs` mount entries](https://github.com/ansible/ansible/pull/41502#issuecomment-443295416) that will make for useful fixtures.

```
c13c1apv7:~ # awk '$3 == "gpfs" { print }' /proc/mounts
/gpfs/automountdir/fs2 /gpfs/automountdir/fs2 gpfs rw,relatime 0 0
fs1 /gpfs/fs1 gpfs rw,relatime 0 0
fs5mpathd /fs5mpathd/a/few/level/mount/point gpfs rw,relatime 0 0
remote /remote gpfs rw,relatime 0 0
/gpfs/automountdir/fs3mpathb /gpfs/automountdir/fs3mpathb gpfs rw,relatime 0 0
/fs4mpathc /gpfs/automountdir/fs4mpathc gpfs rw,relatime 0 0
/gpfs/automountdir/autofs1 /gpfs/automountdir/autofs1 gpfs rw,relatime 0 0
/autofs2 /gpfs/automountdir/autofs2 gpfs rw,relatime 0 0
/autofs3 /gpfs/automountdir/autofs3 gpfs rw,relatime 0 0

c13c1apv7:~ # mount | awk '/type gpfs/ { print }'
/gpfs/automountdir/fs2 on /gpfs/automountdir/fs2 type gpfs (rw,relatime)
fs1 on /gpfs/fs1 type gpfs (rw,relatime)
fs5mpathd on /fs5mpathd/a/few/level/mount/point type gpfs (rw,relatime)
remote on /remote type gpfs (rw,relatime)
/gpfs/automountdir/fs3mpathb on /gpfs/automountdir/fs3mpathb type gpfs (rw,relatime)
/fs4mpathc on /gpfs/automountdir/fs4mpathc type gpfs (rw,relatime)
/gpfs/automountdir/autofs1 on /gpfs/automountdir/autofs1 type gpfs (rw,relatime)
/autofs2 on /gpfs/automountdir/autofs2 type gpfs (rw,relatime)
/autofs3 on /gpfs/automountdir/autofs3 type gpfs (rw,relatime)
```

Reply:
Currently we could just use a local variable in the filter function that has a list of filesystems whose mounts don't always start with `/` that we would like to include.

That could _potentially_ be a use configurable option, but let's just keep it hard coded for now.